### DISCUSSION TITLE
how to sort?

### DISCUSSION THREAD

**Discussion by winner106** [observablehq/plot#1609] [Q&A] [answered]
I would greatly appreciate any help. Thanks!

[https://observablehq.com/d/34d31907b99a4a11](https://observablehq.com/d/34d31907b99a4a11)

**Comment by Fil** [comment, marked as answer]
With the dodgeY transform, the derived *y* channel is not available when sorting the *x* domain, so we can't use the *sort: {x:"y"}* [domain sort mark option](https://observablehq.com/plot/features/scales#sort-mark-option). Instead, you can specify the x domain directly in the top-level options, like so:

`x: {domain: d3.groupSort(transformedData, v => d3.sum(v, d => Math.sqrt(d["150m+"])), d => d.country)},`

**Reply by winner106** [reply]
Thank you so much, Save my life
